### PR TITLE
 Introduce a new module to check that SMM_Code_Chk_En is configured properly.

### PR DESCRIPTION
**Pull request by assafcarlsbad**
Typical output looks like this:

image

### CODE REVIEW

**Review by Sae86 — APPROVED** [review]

File: chipsec/cfg/8086/common.xml [hunk 1]
```diff
@@ -616,6 +616,10 @@ Common (default) XML platform configuration file
       <field name="LOCK"           bit="30" size="1"  desc="Lock (set automatically on the first SMI)" />
       <field name="DEBUG_OCCURRED" bit="31" size="1"  desc="Debug Occurred (set by hardware)" />
     </register>
+    <register name="MSR_SMM_FEATURE_CONTROL" type="msr" msr="0x4E0" desc="Enhanced SMM Feature Control">
```
Comment (BrentHoltsclaw) [inline chipsec/cfg/8086/common.xml:619]: I don't think this is actually a common msr, may want to update it in the configuration files which it applies.

File: chipsec/modules/common/smm_code_chk.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+"""
```
Comment (BrentHoltsclaw) [inline chipsec/modules/common/smm_code_chk.py:1]: GPL header is needed

File: chipsec/modules/common/smm_code_chk.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        self.logger.start_test( "SMM_Code_Chk_En (SMM Call-Out) Protection" )
+
+        results = []
+        for tid in range(self.cs.msr.get_cpu_thread_count()):
+            results.append(self._check_SMM_Code_Chk_En(tid))
+
+        # Check that all CPUs have the same value of MSR_SMM_FEATURE_CONTROL.
+        if not all(_ == results[0] for _ in results):
+            self.logger.log_failed_check( "MSR_SMM_FEATURE_CONTROL does not have the same value across all CPUs" )
+            return ModuleResult.ERROR
```
Comment (BrentHoltsclaw) [inline chipsec/modules/common/smm_code_chk.py:62]:
```suggestion
            return ModuleResult.FAILED
```

File: chipsec/modules/common/smm_code_chk.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        
+        res = results[0] 
+        if res == ModuleResult.ERROR:
```
Comment (BrentHoltsclaw) [inline chipsec/modules/common/smm_code_chk.py:65]: Is this supposed to be FAILED opposed to ERROR?

File: chipsec/cfg/8086/common.xml (hunk 1 above)
Comment (assafcarlsbad) [reply chipsec/cfg/8086/common.xml]: Based on the Intel SDM it seems that the `MSR_SMM_FEATURE_CONTROL` register is available starting from:
* 4th Generation Intel® Core™ Processors (Haswell microarchitecture)
* Atom Processors Based on the Goldmont Microarchitecture

I moved the register definition to the appropriate files. Can you please double-check that I didn't make any mistakes?

**Review by BrentHoltsclaw — APPROVED** [review]

**Review by npmitche — APPROVED** [review]

**Pull request merged**